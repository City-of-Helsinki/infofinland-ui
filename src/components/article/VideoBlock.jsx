import { H3 } from '../Typo'
import Block from '@/components/layout/Block'
import ReactPlayer from 'react-player'
import { useTranslation } from 'next-i18next'

const INITIAL_VOLUME = 0.8
const INITIAL_MUTE = true

const VideoBlock = ({ url, title }) => {
  const { t } = useTranslation('common')
  return (
    <Block className="my-16">
      <H3>{title}</H3>
      <div
        className="flex justify-center pt-2 mt-4 border-t border-gray-lighter aspect-video"
        role="button"
        arial-label={t('video.button')}
      >
        <ReactPlayer
          light
          controls
          volume={INITIAL_VOLUME}
          muted={INITIAL_MUTE}
          url={url}
          height="100%"
          width="100%"
          // onPlay={()=>console.log('on play')}
          // onStart={()=>console.log('start')}
          // onReady={()=>console.log('ready')}
          onError={() => console.error('player error', url)}
        />
      </div>
    </Block>
  )
}

export default VideoBlock
