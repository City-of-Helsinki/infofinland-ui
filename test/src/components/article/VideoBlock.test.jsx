/**
 * @jest-environment jsdom
 */

import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import VideoBlock from '@/components/article/VideoBlock'

describe('VideoBlock', () => {
  it('should render a ReactPlayer with given url  and title inside a Block', () => {
    const { asFragment } = render(
      <VideoBlock
        url="https://icareus-eu1-progressive.secure2.footprint.net/10154/31238760/285903.mp4/1.0.mp4"
        title="Video Title"
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
