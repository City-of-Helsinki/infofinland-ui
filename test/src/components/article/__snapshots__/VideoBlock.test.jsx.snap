// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`VideoBlock should render a ReactPlayer with given url and title inside a Block 1`] = `
<DocumentFragment>
  <div
    class="my-16 ifu-block--article"
  >
    <h3
      class="text-bodytext-color font-sans text-h3 md:text-h3xl"
    >
      Video Title
    </h3>
    <div
      aria-label="video.button"
      class="flex justify-center pt-2 mt-4 border-t border-gray-lighter aspect-video"
      role="button"
    >
      ReactPlayer
    </div>
  </div>
</DocumentFragment>
`;
